credential/base/src/main/java/org/wildfly/security/password/interfaces/RawSaltedSimpleDigestPassword.java

```diff
@@ -56,6 +57,7 @@ public boolean equals(final Object obj) {
             return false;
         }
         RawSaltedSimpleDigestPassword other = (RawSaltedSimpleDigestPassword) obj;
-        return getAlgorithm().equals(other.getAlgorithm()) && Arrays.equals(digest, other.digest) && Arrays.equals(salt, other.salt);
+        return getAlgorithm().equals(other.getAlgorithm()) && MessageDigest.isEqual(digest, other.digest)
+                && MessageDigest.isEqual(salt, other.salt);
```

Comment:
Just a very small comment, we can continue to use `Arrays.equals` for the salt comparison.

Reply:
cool.. will update 👍

Reply:
For reference: reason to not change this is since salt is not user input, it is not susceptible to timing attack